coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/TrinoSqlDialect.java

```diff
@@ -62,4 +74,26 @@ public void unparseIdentifier(SqlWriter writer, SqlIdentifier identifier) {
   public boolean requireCastOnString() {
     return true;
   }
+
+  private void unparseMapOperator(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    writer.keyword(call.getOperator().getName()); // "MAP"
+    final SqlWriter.Frame frame = writer.startList("(", ")");
+    writer.keyword("ARRAY[");
+    // Even numbers are keys
+    for (int i = 0; i < call.operandCount(); i += 2) {
+      writer.sep(",");
+      call.operand(i).unparse(writer, leftPrec, rightPrec);
+    }
+    writer.keyword("], ARRAY[");
+    // Odd numbers are values
+    for (int i = 1; i < call.operandCount(); i += 2) {
+      call.operand(i).unparse(writer, leftPrec, rightPrec);
+      // Last comma is redundant
+      if (i < call.operandCount() - 1) {
+        writer.sep(",");
+      }
+    }
+    writer.keyword("]");
+    writer.endList(frame);
+  }
```

Comment:
Should not we handle the transformation in `Calcite2TrinoUDFConverter`? I think if the operands need transformation, this will not take care of them.

Reply:
Yes, I understand. 
So far, it seems operands  number must be determined because  `CalciteTrinoUDFMap.getUDFTransformer(String calciteOpName, int numOperands)`. However `map` is variable length parameter, I can't put it in `UDF_MAP`.
Do we plan to support functions with variable length parameters? I thinks this is meaningful topic.

Reply:
Agreed. Maybe we can introduce `getUDFTransformer(String calciteOpName)` that means the `Op` takes variable length parameters?